server/api/dev.js

```diff
@@ -0,0 +1,214 @@
+        {
+          user: 'dev', name: 'Approved example', submission_status: 'approved', modified_by: 'VertigoOne',
+        },
+        {
+          user: 'dev', name: 'Rejected example', submission_status: 'rejected', modified_by: 'VertigoOne', rejection_reason: 'Rejected because.',
+        },
+        { user: username, name: 'Pending example', submission_status: 'pending' },
+        { user: username, name: 'Withdrawn example', submission_status: 'withdrawn' },
+        {
+          user: username, name: 'Approved example', submission_status: 'approved', modified_by: 'VertigoOne',
+        },
+        {
+          user: username, name: 'Rejected example', submission_status: 'rejected', modified_by: 'VertigoOne', rejection_reason: 'Rejected because.',
+        },
+      ];
+
+      // add sequential dev# for id (NOTE: zero-indexed, can do i+1 if desired)
+      entryVersions = entryVersions.map((v, i) => ({ ...v, id: `dev${i}`, background_color: BACKGROUND_COLORS[i % 3] }));
+
+      const existingEntries = await db.select(`SELECT width, height, url FROM entries WHERE height > 600 AND POSITION('https://firebasestorage.googleapis.com' IN url) > 0 LIMIT ${entryVersions.length}`);
+      await db.insert('entries', entryVersions.map((version, i) => ({
+        ...existingEntries[i], ...version,
+      })));
+
+      /*
+        ------- CONTEST_ENTRIES -------
+        Connect the two
+      */
+      await db.insert('contest_entries', entryVersions.map(({ id }, i) => ({
+        contest_id: 'dev',
+        entry_id: id,
+        rank: null,
+        category: categories[i % categories.length].category, // alternating entries
+      })));
+
+      /*
+        ------- VOTES -------
+        Random voting, all users on all flags
+      */
+      const approvedEntries = entryVersions.filter((version) => version.submission_status === 'approved');
+
+      await db.insert('votes', users.flatMap(({ username: u }) => (
+        approvedEntries.map(({ id }) => ({
+          username: u,
+          entry_id: id,
+          contest_id: 'dev',
+          rating: u === approvedEntries.user
+            ? VOTE_MAX
+            : Math.floor(Math.random() * (VOTE_MAX + 1)),
+          last_modified: NOW,
+        }))
+      )));
+    } else {
+      const contest = getDevContest(status);
+      // Switch contest status
+      if (isExisting) {
+        // update rejects date for some reason
+        // await db.update('contests',
+        // [contest], Object.keys(contest).map((key) => (key === 'id' ? '?id' : key)));
+        await db.any(`UPDATE contests
+          SET
+            name = '${contest.name}',
+            date = '${contest.date}',
+            year_end = ${contest.year_end},
+            submission_start = '${contest.submission_start}',
+            submission_end = '${contest.submission_end}',
+            vote_start = '${contest.vote_start}',
+            vote_end = '${contest.vote_end}'
+          WHERE id = 'dev';`);
```

Comment:
Looks like this caused a code scanning alert: `Database query built from user-controlled sources`. Using `db.update` should avoid this.

Reply:
Can you play around with update on your end? I just couldn't get past the update issue.

Reply:
Sure, I realized there isn't an update that currently puts a date, so I was just speculating.